src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -71,13 +71,20 @@ public function handle(GetResponseEvent $event)
         $request = $event->getRequest();
         $session = $request->hasPreviousSession() ? $request->getSession() : null;
 
-        if (null === $session || null === $token = $session->get($this->sessionKey)) {
+        if (null === $session || null === $serializedToken = $session->get($this->sessionKey)) {
             $this->tokenStorage->setToken(null);
 
             return;
         }
 
-        $token = unserialize($token);
+        try {
+            $token = $this->unserialize($serializedToken);
+        } catch (\ErrorException $e) {
+            if (null !== $this->logger) {
+                $this->logger->warning('Failed to unserialize the security token from the session.', array('key' => $this->sessionKey, 'received' => $serializedToken, 'exception' => $e));
```

Comment:
Shouldn't this be an error instead? It means that something goes wrong somewhere.

Can you rename `key` to `session_key`? This should describe the function a bit more when viewing the logs. At first I thought this was the firewall (context) key, but that's included in the session key.

Reply:
Can do, though I was keeping it in-line with https://github.com/symfony/symfony/blob/96e53f8837255d4f9ad3c7d2c68dcda1abbb0cb5/src/Symfony/Component/Security/Http/Firewall/ContextListener.php#L109

Reply:
Fair enough, best to keep it consistent then.